File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/HookCollector.java
package cn.tongdun.mobrisk.core.collectors;

import cn.tongdun.mobrisk.core.utils.JNIHelper;

/**
 * @description: detect hook
 * @author: wuzuchang
 * @date: 2023/4/19
 */
public class HookCollector {

    public HookCollector() {
    }

    public String detectHook() {
        return JNIHelper.detectHook();
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/HookCollector.kt
package cn.tongdun.mobrisk.core.collectors

import cn.tongdun.mobrisk.core.tools.JNIHelper
import cn.tongdun.mobrisk.core.tools.executeSafe

/**
 * @description:
 * @author: wuzuchang
 * @date: 2023/5/15
 */
interface HookInterface {
    fun detectHook(): String
}

class HookCollector : HookInterface {
    override fun detectHook(): String = executeSafe({ JNIHelper.call1() }, "")
}